### PR TITLE
Fixed space followed by uppercase to be a single dash.

### PR DESCRIPTION
**Pull request by jeffreypriebe**
Currently, dashify will convert `'foo Bar'` to `'foo--bar'`.

Changed to output a single dash.

Added test, changed regex, and updated README.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ module.exports = function dashify(str) {
   if (typeof str !== 'string') {
     throw new TypeError('dashify expects a string.');
   }
-  str = str.replace(/[A-Z]/g, '-$&');
+  str = str.replace(/([a-z]{1})([A-Z])/g, '$1-$2');
```
Comment (jonschlinkert) [inline index.js]: great thanks. we don't need the `{1}` though, just the `[a-z]` should be enough.

Comment (jeffreypriebe) [inline index.js]: Indeed, should work either way. Will modify and squash commit into this.

Comment (jonschlinkert) [inline index.js]: great, thanks!

**Comment by jonschlinkert** [comment]
thanks again! published to npm and [released as 0.2.1](https://github.com/jonschlinkert/dashify/releases/tag/0.2.1)

**Comment by jeffreypriebe** [comment]
:+1:

**Pull request merged**